### ISSUE TITLE
Inconsistent co-ordinates from Events and to methods on Float Canvas

### ISSUE THREAD

**Issue by TonyFlury** [wxWidgets/Phoenix#1825]
**Operating system**:                         Ubuntu 19.10
**wxPython version & source**:         pypi 4.10
**Python version & source**:            stock

**Description of the problem**:
It is notable when using FloatCanvas that the various co-ordinate systems used seem to inconsistent and not well documented.

Example : In the absence of the ScalePixeltoWorld and similar methods, the Draw objects used a co-ordinate system based on a zero, zero point in the center of the canvas. The MouseEvents use Windowing co-ordinate system (where top right seems to be zero, zero). The Documentation isn't clear though as to which method receive data in which co-ordinate systems.

Also - using the RubberbandBox,, it states that the Callback function received a wx.Rect object; The documentation for the wx.Rect states that the width and height will never be zero; however this is not true - dragging the Rubberband box down the Height is negative, and dragging the box left results in a negative width.

I am not suggesting that the code has bugs (obviously changing co-ordinate systems will break existing code), but I do think the documentation can be more clear in terms for the co-ordinate systems.

**Comment by RobinD42** [comment]
I think the only operations on a `wx.Rect` that will protect against dimensions <0 are the `Inflate` and `Deflate` methods. Everything else is a free-for-all.

For the co-ordinate systems or other issues, I don't think @PythonCHB is putting much effort lately in maintaining FloatCanvas, so PRs for fixes or improvements are welcome.

**Comment by TonyFlury** [comment]
Robin, I am not suggesting code changes, but the documentation does need clarification.

**Comment by RobinD42** [comment]
My statement applies to documentation PRs too.

**Comment by PythonCHB** [comment]
Thanks for the ping. And yes, Documentation PRs are gladly accepted!

It's true that I haven't done much for/with FloatCanvas lately, but I would like to support it as best I can.

In regards to your questions, some clarifications, and maybe you could figure out how to make it clear in the documentation:

> It is notable when using FloatCanvas that the various co-ordinate systems used seem to inconsistent and not well documented.

Not well -- documented -- yes, documentation is hard, and not as interesting as coding :-(

> Example : In the absence of the ScalePixeltoWorld and similar methods, the Draw objects used a co-ordinate system based on a zero, zero point in the center of the canvas.

well, not quite -- the DrawObjects use the "world" coordinate system -- which is floating point, rather than integer (hence the name of the canvas) and arbitrarily related to the pixel coords of the Window. the relationship is based on the projection set, and where the origin is is based on the current viewport.

> The MouseEvents use Windowing co-ordinate system (where top right seems to be zero, zero).

Indeed -- that's what we get from the system -- MouseEvents are, indeed, in pixel coords (though top left is (0, 0) -- But the FloatCanvas derived ones also provide a way to get the world coords attribute as well. 

> The Documentation isn't clear though as to which method receive data in which co-ordinate systems.

As for the Mouse events, here's the docstring:

```
class _MouseEvent(wx.PyCommandEvent):

    """
    This event class takes a regular wxWindows mouse event as a parameter,
    and wraps it so that there is access to all the original methods. This
    is similar to subclassing, but you can't subclass a wxWindows event
    The goal is to be able to it just like a regular mouse event.
    It adds the method:
    GetCoords() , which returns an (x,y) tuple in world coordinates.
    Another difference is that it is a CommandEvent, which propagates up
    the window hierarchy until it is handled.
    """
```

so GetCoords() will give you world coords. I fi were to write that today, I'd probably make it a property.

As for what coordinate system is expected in various methods, Im welcome to suggestions as to where we need more/better docs.

> Also - using the RubberbandBox,, it states that the Callback function received a wx.Rect object; The documentation for the wx.Rect states that the width and height will never be zero; however this is not true - dragging the Rubberband box down the Height is negative, and dragging the box left results in a negative width.

I can't find those docs right now, but I think Robin may have answered: wxRect can be negative, and that's actually very helpful for FloatCanvas and Rubber Band box -- you need to be able to have a Rect in either direction!

**Comment by TonyFlury** [comment]
I agree documentation is tough, but critical - and so confusing when it isn't clear.
I will have a look at the documentation set and see if I can do some small hits to make it clearer - I will mark the commits with this PR.

Is there a contributor guide I should be following ?

**Comment by PythonCHB** [comment]
Thanks -- anything you can do is appreciated.

Not sure about the contributor's guide: @RobinD42 ?

But for FloatCanvas specific code, I'll look at any PR you can do.
